### PR TITLE
Create dart.yml for CI 

### PR DESCRIPTION
**Pull request by cameronaaron**
I am starting inital work on integrating CI for our project to do things such as automatic linting and error detection on pull. This might not be ready to actually be pushed yet because I am still adding rules but wanted to open up a PR for comments

### CODE REVIEW

**Comment by tinataleb** [comment]
Thanks for starting this, it'll help us move forward faster with an ease of mind!

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Close Stale Issues
+uses: actions/stale@1.0.0
+  with:
+    # Token for the repo. Can be passed in using {{ secrets.GITHUB_TOKEN }}
+    repo-token: 
+    # The message to post on the issue when tagging it. If none provided, will not mark iusses stale.
```
Comment (Kthulu120) [inline .github/workflows/stale.yml]: Seems like your required to have token for repo and perhaps changing this to would fix it proceeded by adding a github secret to repo.
```
repo-token:  {{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    container:
+      image:  google/dart:latest
```
Comment (Kthulu120) [inline .github/workflows/dart.yml]: for some reason flutter sdk isn't installed on this container or if it is its not in the PATH so you'll need to add the flutter sdk. [This](https://gist.github.com/Kthulu120/59639e3a81dc8e61f37c78177d39ac20) is the Dart Action that I use to automatically produce builds for flutter on master branch. You'll need to change it to all branches besides that as long as the tests pass this config will work. Just a option.

File: .github/workflows/fluttertests.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+name: Dart/Flutter Package Analyzer
+uses: axel-op/dart-package-analyzer@v2.0.0
```
Comment (Kthulu120) [inline .github/workflows/fluttertests.yml]: This has to be wrapped by an `on:`  attribute from what the github workflow is telling me.

**Review by Kthulu120 — CHANGES_REQUESTED** [review]
Overall good start, now just need to modify and test these workflows on your branch.

**Comment by Kthulu120** [comment]
Checking in hows progress?

**Comment by cameronaaron** [comment]
@Kthulu120 Thank you this passes and can be merged

**Pull request merged**